numpy/core/tests/test_multiarray.py

```diff
@@ -2438,6 +2438,15 @@ def test_dot_out_mem_overlap(self):
             assert_raises(ValueError, np.dot, a, b, out=b[::2])
             assert_raises(ValueError, np.dot, a, b, out=b.T)
 
+    def test_dot_zero_strides(self):
+        a = np.zeros(1, 'f')
+        a = np.broadcast_to(a, (2))
+        b = np.zeros((2, 2), 'f')
+        c = a.dot(b)
+        assert_equal(c, [0, 0])
+        d = b.dot(a)
+        assert_equal(d, [0, 0])
```

Comment:
Also probably should do a `for dtype in dtypes:` loop like in the other tests for completeness sake.